src/box/lua/iproto.c

```diff
@@ -28,240 +28,119 @@
 
 struct mh_strnu32_t *iproto_key_translation;
 
-/**
- * IPROTO constant from `src/box/iproto_{constants, features}.h`.
- */
-struct iproto_constant {
-	/**
-	 * Constant literal, name of constant.
-	 */
-	const char *const name;
-	/**
-	 * Constant literal, value of constant.
-	 */
-	const lua_Integer val;
-};
-
 /**
  * Pushes an array of IPROTO constants onto Lua stack.
  */
 static void
-push_iproto_constant_subnamespace(struct lua_State *L, const char *subnamespace,
-				  const struct iproto_constant *constants,
-				  int constants_len)
+push_iproto_constant_subnamespace(
+	struct lua_State *L, const char *subnamespace,
+	const struct iproto_constant *constants, int constants_len,
+	const char *strip_prefix)
 {
+	size_t prefix_len = 0;
+	if (strip_prefix != NULL)
+		prefix_len = strlen(strip_prefix);
 	lua_createtable(L, 0, constants_len);
 	for (int i = 0; i < constants_len; ++i) {
-		lua_pushinteger(L, constants[i].val);
-		lua_setfield(L, -2, constants[i].name);
+		const char *name = constants[i].name;
+		int value = constants[i].value;
+		if (strstr(name, "RESERVED"))
+			continue;
+		assert(strncmp(name, strip_prefix, prefix_len) == 0);
```

Comment:
Passing `NULL` to `strncmp` (in case `strip_prefix` is `NULL`) is UB:
> The behavior is undefined when either lhs or rhs is the null pointer.

[^1]: https://en.cppreference.com/w/c/string/byte/strncmp

Reply:
Even with `count = 0`. Great!
Thanks, I will fix it.

Reply:
Fixed